### ISSUE TITLE
Add “Show password”  fuctionality to password field

### ISSUE THREAD

**Issue by hidde** [mozilla-iam/auth0-custom-lock#71]
As suggested by https://discourse.mozilla.org/t/changing-the-look-of-your-login-part-2/26134/2?u=hidde

> In particular on mobile, I would really appreciate a “show password” box.
> Nothing more frustrating than getting 14 characters into a 20 character password and then going “Oops… Did I hit that key once… or twice?”.

**Comment by hidde** [comment]
Personal opinion: this would be better as a platform feature. And it may proof difficult to implement in a way that does not interfere with password managers.

**Comment by hidde** [comment]
@gene1wood Built-in to the browser. Imho if this is a feature people need, I think they'd need it for all password fields always.

**Comment by hidde** [comment]
Closing this issue as there is currently no work planned on this.

**Comment by gene1wood** [comment]
Currently, since this is not a feature in any browser, this functionality is implemented on websites themselves. 

Edge has a [non-standard builtin facility for this](https://developer.mozilla.org/en-US/docs/Web/CSS/::-ms-reveal)

And the js and css to enable this feature is [extremely simple](https://www.w3schools.com/howto/howto_js_toggle_password.asp)

> it may proof difficult to implement in a way that does not interfere with password managers.

I disagree. Since the point of time that the textbox would be changed from type `password` to type `text` would be after the page had rendered (since it's a manual user interaction), the password manager would have long ago entered the password into the field, and I know of no password managers that remove your password when the field type changes in the DOM.

@hidde Is the reason for closing this that it would be too hard to do?

**Comment by hidde** [comment]
> I disagree. Since the point of time that the textbox would be changed from type password to type text would be after the page had rendered (since it's a manual user interaction), the password manager would have long ago entered the password into the field, and I know of no password managers that remove your password when the field type changes in the DOM.

I retract my original statement, you're right, it would not impact password managers as I said it would, because it happens after page is rendered and after password manager magic happens. 

> @hidde Is the reason for closing this that it would be too hard to do?

Yes and no. You're absolutely right, JavaScript-wise it is trivial to implement. 

But I reckon this is still the kind of addition that would need some thought around UX (maybe it would add more clutter to the interface? (cc @m-branson )), design (can the button fit on a small screen and will it then have enough touch affordance), accessibility (will a change of input have unexpected behaviour in screenreaders?), implementation (ok, trivial), QA and deploying it… it would cost time that I will not have, especially with Mozillians Repositioning coming up.

Also, I believe use of password managers will be more and more encouraged for Mozillians, and those of us using password managers would likely not need to visibly see the password? If that's the case, that would be another reason to not prioritise for it at this time.

So basically, reason for closing is that it currently does not fit with available time and product strategy.

**Comment by hidde** [comment]
I'll close this again so that our issue list stays up to date, of course feel free to reopen if the above doesn't sound right.

**Comment by gene1wood** [comment]
I'd like to revisit the idea brought up a couple days ago of closing issues that we want to work on but don't have resources right now. I'm going to re-open this and remove `wontfix` because I don't think it's a good description of this issue.

https://discourse.mozilla.org/t/process-improvement-proposal-close-backlog-issues-with-wontfix/29731/3
